misc/python/materialize/checks/all_checks/retain_history.py

```diff
@@ -0,0 +1,87 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from textwrap import dedent
+
+from materialize.checks.actions import Testdrive
+from materialize.checks.checks import Check
+from materialize.checks.executors import Executor
+from materialize.mz_version import MzVersion
+
+
+class RetainHistory(Check):
+    def _can_run(self, e: Executor) -> bool:
+        return e.current_mz_version >= MzVersion.parse_mz("v0.81.0")
+
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+                $ postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
+                ALTER SYSTEM SET enable_logical_compaction_window = true
```

Comment:
Any reason we don't just enable this for all testing?

Reply:
Yes, makes sense to do that.

Reply:
✅ Done.